### PR TITLE
Fix: Containers/InitContainers

### PR DESCRIPTION
**Pull request by diamonwiggins**
* Fixes a bug that resulted in charts failing to render when multiple containers were configured on a single app
* https://github.com/helm/helm/issues/7149

### CODE REVIEW

File: charts/replicated-library/templates/lib/_container.tpl [hunk 1]
```diff
@@ -101,5 +101,5 @@
     {{- toYaml . | nindent 4 }}
   {{- end }}
   {{- end -}}
-{{- end -}}
+{{ end }}
```
Comment (chris-sanders) [inline charts/replicated-library/templates/lib/_container.tpl:104]:
```suggestion
{{- end }}
```

File: charts/replicated-library/templates/lib/_container.tpl [hunk 2]
```diff
@@ -7,7 +7,7 @@
     {{- fail "_container.tpl requires the 'app' ContextValues to be set" -}}
   {{- end -}}
   {{- $_ := set $.ContextValues "names" (dict "context" "app") -}}
-{{- range $containerName, $containerValues := $values.containers -}}
+{{ range $containerName, $containerValues := $values.containers }}
```
Comment (chris-sanders) [inline charts/replicated-library/templates/lib/_container.tpl:10]:
```suggestion
{{- range $containerName, $containerValues := $values.containers }}
```

File: charts/replicated-library/templates/lib/_init_container.tpl [hunk 3]
```diff
@@ -7,7 +7,7 @@
     {{- fail "_init_container.tpl requires the 'app' ContextValues to be set" -}}
   {{- end -}}
   {{- $_ := set $.ContextValues "names" (dict "context" "app") -}}
-{{- range $containerName, $containerValues := $values.initContainers -}}
+{{ range $containerName, $containerValues := $values.initContainers }}
```
Comment (chris-sanders) [inline charts/replicated-library/templates/lib/_init_container.tpl:10]:
```suggestion
{{- range $containerName, $containerValues := $values.initContainers }}
```

File: charts/replicated-library/templates/lib/_init_container.tpl [hunk 4]
```diff
@@ -65,5 +65,5 @@
   resources:
     {{- toYaml . | nindent 4 }}
   {{- end }}
-{{- end -}}
+{{ end }}
```
Comment (chris-sanders) [inline charts/replicated-library/templates/lib/_init_container.tpl:68]:
```suggestion
{{- end }}
```

**Review by chris-sanders — CHANGES_REQUESTED** [review]
I haven't had time to dig through all of this, but just removing the end line chomping will likely solve the problem without leaving random spaces throughout the template. Can you try this with just the beginning and not the end `-` and see if that works?

**Review by chris-sanders — APPROVED** [review]
Great looks good to me. Generally speaking leading `-` are most of the time ok and trailing `-` are bad news unless it's purely logic. This reads more like I would expect to see it, if it's working for you I don't need more testing on other ways it could be written.

**Comment by diamonwiggins** [comment]
> Great looks good to me. Generally speaking leading `-` are most of the time ok and trailing `-` are bad news unless it's purely logic. This reads more like I would expect to see it, if it's working for you I don't need more testing on other ways it could be written.

Yup, makes sense. I probably need to do a second pass of the chart to make things consistent here and ensure I don't have other bugs.

**Pull request merged**